### PR TITLE
Remove unneeded index.ts files in querying

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/47147
Follow up for https://github.com/metabase/metabase/pull/47160

Removes unneeded re-exports.

### CODE REVIEW

**Review by uladzimirdev — APPROVED** [review]
😍

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Filter.tsx [hunk 1]
```diff
@@ -2,12 +2,10 @@ import cx from "classnames";
 
 import { useInteractiveQuestionContext } from "embedding-sdk/components/private/InteractiveQuestion/context";
 import CS from "metabase/css/core/index.css";
-import {
-  FilterModalBody,
-  FilterModalFooter,
-  FilterModalHeader,
-  useFilterModal,
-} from "metabase/querying/filters";
+import { FilterModalBody } from "metabase/querying/filters/components/FilterModal/FilterModalBody";
```
Comment (kamilmielnik) [inline enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Filter.tsx:5]: Just my 2 cents.

`index.ts` files are a great way of encapsulating things in modules.
If we allow ourselves to import anything from anywhere (like here, these imports looks rather _intimate_ now) then we're basically giving up on module encapsulation. Developers won't think twice about how to structure things. Nothing will indicate that it's not ok to e.g. import a CSS file of a sub-sub-sub-component of a module.

The graph of dependencies between files will have more edges now.

It's good to have this `metabase/querying/filters` module and to control what is available outside it from the `index.ts` file. If you want to import something from this module and it's not exported from it, it will at least make you think: "hey, should I really use it? Maybe it should live elsewhere if it needs to be reused outside that module?"

It'd be worth discussing what constitutes a "module" (is it `metabase/querying`, or `metabase/querying/filters`, or `metabase/querying/filters/components`). But regardless of what it is, if we lose the notion of a module, then it will make it more difficult for us to make the codebase modular (which I thought is one of our goals).

tl;dr I don't support this change.

Comment (ranquild) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Filter.tsx:5]: Using re-exports was initially my idea and now I believe it was a mistake. People didn't maintain this and we had many direct imports anyway.
